### ISSUE TITLE
Auto view on new-tab.

### ISSUE THREAD

**Issue by doyousketch2** [Quicksaver/OmniSidebar#142]
Hey, I love your add-on.
I like the fact that it ducks my bookmarks out of view when they aren't being used.

I think the only thing I'd like to see, is an option:
automatically pop into view when I click the + sign to get a new-tab.

I would probably only check this on the main-sidebar,
which shows my bookmarks;  not the twin-sidebar, which shows history.

**Comment by Quicksaver** [comment]
Why do you need this? You could just middle-click or ctrl+left-click on a bookmark to open it in a new tab.

**Comment by doyousketch2** [comment]
I don't permanently keep the sidebar open.
The point is that it only shows when I need it,
and I need it when I open a new tab.

So it would be nice if the sidebar automatically showed up when I open a new-tab,
instead of having to click the side of the screen every time.

**Comment by Quicksaver** [comment]
My question was why do you need it in new tabs? I assumed you want to open some bookmarked page in that new tab instead of replacing the current tab. As I mentioned, you can do that just as well by middle-clicking a bookmark in the sidebar (which opens that bookmark in its own tab) instead of opening an empty new tab first. It takes the same amount of clicks and is a more efficient action even.

Is this not what you're trying to do?

**Comment by doyousketch2** [comment]
I've deleted my bookmarks bar up-top.

I now use your sidebar as my bookmarks, but it is always hidden, unless I click the left-border.

When I open a new tab, I don't see my bookmarks, unless I scroll over and click the left-border.

It would be nice if the sidebar would just open up on its own when I bring up a new blank tab.

**Comment by Quicksaver** [comment]
That still wasn't my question. Why do you need the bookmarks in the new tab? What do you want to do with the bookmarks in the new tab that you can't do in any other tab?

**Comment by doyousketch2** [comment]
<rolls eyes>  I've already stated why.

**Comment by Quicksaver** [comment]
No, you've stated what is happening now and what you want to happen instead. I really don't know how to more clearly put the questions that I asked in my previous comment...

Here's the thing. the new tab page and the bookmarks sidebar (or any other sidebar that I know of even) have virtually nothing to do with each other. There's no reason for why the (bookmarks) sidebar should or needs to interact with the new tab page. So a feature like that really has no place in an add-on directed only for sidebar enhancements.

That's why I'm trying to understand what exactly you're looking for here... Why do you want the sidebar to appear when opening new tabs? What do you want to do with it at that time that you can't already do from before (and even without) opening the new tab? It may seem obvious to you, but remember that not all users have the same browsing habits; clearly yours are so different from mine that I can't even envision what you're trying to do.

**Comment by doyousketch2** [comment]
I feel like this is a pointless conversation.

I don't need to re-state my idea 3 times to clarify anything.
I've already stated the desired outcome, and why it would be used.

**Comment by EMLVIRUS** [comment]
I believe what @doyousketch2 want would be:

> When I open a new tab, I don't see my bookmarks, unless I scroll over and click the left-border.
It would be nice if the sidebar would just open up on its own when I bring up a new blank tab.

He doesn't keep his sidebar opened, and because he has closed bookmarks bar on the top, he would like the sidebar to show up itself whenever there is a new tab so that he can choose which bookmark to be opened.
 Am I right?

**Comment by Quicksaver** [comment]
> I've already stated the desired outcome, and why it would be used.

@doyousketch2 I'm sorry, but again you didn't. You did state the desired outcome, and also what is happening currently. But you have not stated why you want that desired outcome.

> so that he can choose which bookmark to be opened.

@EMLVIRUS This is exactly what I haven't seen clarified yet. It's the answer to "why you would need the sidebar" that I'm looking for and what I'm trying to understand.

> it be really inconvenient if you don't keep the sidebar opened as you have to make it show up yourself by moving your mouse frequently.

Isn't it more inconvenient to have to open the new tab first? You need to click on a button next to the tabs, which can be pretty small by itself (especially if tabs are overflowing, which makes it even smaller) or even move around the top area of the window if you don't have enough tabs to fill that area.

The target to open the sidebar is unquestionably larger and much easier to get to (i.e. move the mouse indiscriminately to the margin, rather than pointing it precisely to the new tab button, it also has a customizable keyboard shortcut as well which the new tab button has a fixed one). The number of actions in both cases is exactly the same as well so there's no loss in workflows there. The speed with which/time taken to open the sidebar will undoubtedly be faster than that to open the new tab.

Also, what I meant in my last post is that the bookmarks sidebar is virtually unrelated to the new tab. Whenever you open a new tab, there's no guarantee you actually want to use any of the bookmarks, you may want to just type an address or run a web search. So in those cases you would see the sidebar appear uselessly. Not to mention you would also be loading the new tab page uselessly as well. Whereas if your starting point is the sidebar, you can choose to open in a new tab or existing one from there (there's also Tab Mix Plus which lets you open bookmarks in new tabs automatically through left-clicks directly).

Why not instead of opening the new tab and then choosing a bookmark to load, open the bookmarks sidebar instead (still a single action) and then open whatever bookmark in a new tab from it? That way, you only see and load the bookmarks sidebar when you actually will use it, you still perform the same number of actions (again, with the action to open the sidebar being much more accessible and configurable than the action to open a new tab), and you don't uselessly load the new tab page when you won't use any of its contents (because you intend to load a bookmark).

This is what I'm trying to understand, where's the gain in opening the new tab first? Note that I'm not looking for you to convince me to agree to using that behavior myself, just to help me understand it.
